text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,1119 @@
+```
+
+When target is `linux/amd64`
+  - `pack` will determined **target root binary folder** is `./linux/amd64` 
+    - `pack` will copy everything from the `./linux/arm64` into the base buildpack package folder, because there are no conflicts the expected output will be:
+
+```bash
+
+.
+└── cnb
+    └── buildpacks
+        └── {ID}
+            └── {version}
+                ├── bin
+                │    ├── build
+                │    ├── detect
+                │    └── foo         // specific platform binary
+                ├── buildpack.toml
+                ├── foo              // common file
+                └── package.toml
+```
+
+On the other hand, When target is `linux/arm64`
+- `pack` will determined **target root binary folder** is `./linux/arm64`
+  - `pack` will copy everything from the `./linux/arm64` into the base buildpack package folder, because there is a conflict trying to copy `./linux/arm64/foo` into `cnb/buildpacks/{ID}/{version}/foo`, `pack` will override it and moves forward, resulting into:
+
+```bash
+.
+└── cnb
+    └── buildpacks
+        └── {ID}
+            └── {version}
+                ├── bin
+                │    ├── bar        // specific platform binary
+                │    ├── build
+                │    └── detect
+                ├── buildpack.toml
+                ├── foo             // override files with the content of the arm64 folder
+                └── package.toml
+```
+
+#### Buildpacks authors do not use targets AND `platform.os` is not present at `package.toml`
+
+This seems to be the case for [Paketo Buildpacks](https://github.com/paketo-buildpacks/java/blob/main/package.toml) 
+or [Heroku](https://github.com/heroku/buildpacks-jvm/blob/main/meta-buildpacks/java/package.toml), and it represents how `pack` 
+will work for most the users when new behavior is implemented
+
+A simplified version of Buildpack Authors folder structures is:
+
+```bash
+├── bin
+│ ├── build
+│ └── detect
+├── buildpack.toml
+└── package.toml
+```
+
+In these cases: We expect `pack` to keep doing what is doing today, but with the warning messages we mentioned above to 
+let end users know things are changing. 
+
+```bash
+pack buildpack package <buildpack> --config ./package.toml --publish 
+Warning: A new '--target' flag is available to set the platform for the buildpack package, using 'linux' as default
+Successfully published package <buildpack> and saved to registry
+
+# Or
+pack buildpack package  <buildpack> --config ./package.toml --format file 
+Warning: A new '--target' flag is available to set the platform for the buildpack package, using 'linux' as default
+Successfully created package <buildpack> and saved to file
+ ```
+**Output**: pack will create a buildpack package image (as it is doing it today) with the provided binaries and a 
+[configuration](https://github.com/opencontainers/image-spec/blob/main/config.md#properties) with the following target 
+platform:
+
+```json
+{
+  "architecture": "",
```

Comment:
This is how it works today, but this feels like a bug. The OCI image spec says this field is required.

Reply:
Maybe it is a bug, but apparently nobody is complaining about it! I didn't want to start filling this attribute and break the current behavior for end-users.

Reply:
Actually this did come up recently ([here](https://github.com/buildpacks/pack/issues/1968) and [here](https://cloud-native.slack.com/archives/C0331B61A1Y/p1706546411819879)) I __think__ we said this RFC would fix those cases but maybe we assumed first the buildpack author would upgrade to use `targets`. In any case, it probably doesn't hurt... I'm fine to leave it out of this RFC though

Reply:
Yeah, I think is what @jericop is trying to fix also [here](https://github.com/buildpacks/pack/pull/2080)